executor/instruction/mod.rs

```diff
@@ -358,29 +365,96 @@ impl<T: Hkt> Checker<T> {
+                        let rhs_variable_value = Self::get_vertex_value(rhs, row.as_ref(), &context.parameters);
+                        let rhs_value =
+                            Self::read_value(context.snapshot.as_ref(), &context.thing_manager, &rhs_variable_value)?;
+                        // let rhs_value = Self::read_value(context.snapshot.as_ref(), &context.thing_manager, &rhs_variable_value)?;
+                        if let Some(rhs_value) = rhs_value {
+                            let comp_range = match comparator {
+                                Comparator::Equal => (Bound::Included(rhs_value.clone()), Bound::Included(rhs_value)),
+                                Comparator::Less => (Bound::Unbounded, Bound::Excluded(rhs_value)),
+                                Comparator::LessOrEqual => (Bound::Unbounded, Bound::Included(rhs_value)),
+                                Comparator::Greater => (Bound::Excluded(rhs_value), Bound::Unbounded),
+                                Comparator::GreaterOrEqual => (Bound::Included(rhs_value), Bound::Unbounded),
+                                Comparator::Like => continue,
+                                Comparator::Contains => continue,
+                                Comparator::NotEqual => continue,
+                            };
+                            range = intersect(range, comp_range);
+                        }
+                    } else {
+                        debug_assert!(
+                            rhs.as_variable().expect("RHS of comparison must be a variable") == target_variable
+                        );
+                        let lhs_variable_value = Self::get_vertex_value(lhs, row.as_ref(), &context.parameters);
+                        let lhs_value =
+                            Self::read_value(context.snapshot.as_ref(), &context.thing_manager, &lhs_variable_value)?;
+                        if let Some(lhs_value) = lhs_value {
+                            let comp_range = match comparator {
+                                Comparator::Equal => (Bound::Included(lhs_value.clone()), Bound::Included(lhs_value)),
+                                Comparator::Less => (Bound::Excluded(lhs_value), Bound::Unbounded),
+                                Comparator::LessOrEqual => (Bound::Included(lhs_value), Bound::Unbounded),
+                                Comparator::Greater => (Bound::Unbounded, Bound::Excluded(lhs_value)),
+                                Comparator::GreaterOrEqual => (Bound::Unbounded, Bound::Included(lhs_value)),
+                                Comparator::Like => continue,
+                                Comparator::Contains => continue,
+                                Comparator::NotEqual => continue,
+                            };
+                            range = intersect(range, comp_range);
+                        }
+                    }
                 }
                 _ => (),
```

Comment:
Can we also add the `Is` check as if it were equality here? It's probably less common for attributes, but still.